### PR TITLE
Day 2 solution with test in java

### PR DESCRIPTION
**Pull request by malhotra1432**
Please make sure you have done the following:

- [ ] Created a folder with the name of language you code inside `/Day<n>/` directory (Where n is for nth day)
- [ ] Added your Codes
- [ ] Tested your codes
- [ ] Added credits (author and date) at the top of the code
- [ ] Updated README.md
- [ ] Added yourself as a contributor using `all-contributors-cli`

### CODE REVIEW

File: Day3/Java/HammingDistancePrabhat.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * Solution
+ */
+public class HammingDistancePrabhat {
+
+    public int calculateHammingDistance(String firstString, String secondString){
+        int totalChange = 0;
+        byte[] aByteArray = firstString.getBytes();
+        byte[] bByteArray = secondString.getBytes();
+
+        for(int i = 0; i < aByteArray.length; i++){
```
Comment (aaditkamat) [inline Day3/Java/HammingDistancePrabhat.java:17]: You need to first check if the strings are of the same length. Hamming distance is undefined for strings of different lengths.

File: Day3/README.md [hunk 2]
```diff
@@ -388,7 +388,49 @@ public class HammingDistance {
 
 }
 
+## Solution with test
```
Comment (aaditkamat) [inline Day3/README.md:391]: Add your solution under the correct section 'Java Implementation' in the README.

File: Day2/README.md [hunk 3]
```diff
@@ -904,6 +904,68 @@ public class StringReverseAndPalin {
 	}
 
 }
+```
```
Comment (aaditkamat) [inline Day2/README.md:907]: Please add this under the section `Java Implementation` of the README.

File: Day2/Java/AstringReversalPrabhat.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    if (reversedString.equals(randomString)) {
+      System.out.println(randomString + " is a palindrome.");
+    } else {
+      System.out.println(randomString + " is not a palindrome.");
+    }
+  }
+}
+
+/*
+ * Test A - String Reversal Test
```
Comment (aaditkamat) [inline Day2/Java/AstringReversalPrabhat.java:49]:
```suggestion
 * Test B - Palindrome Check Test
```

**Review by aaditkamat — CHANGES_REQUESTED** [review]
You've created a PR again from your master branch, but at least it does not introduce any additional changes that are not required. Please amend your docs and code as per the suggestions that I have given in the comments.

**Comment by malhotra1432** [comment]
@aaditkamat Sorry about mistakes. I'll close this PR and request a new PR with proper work.

**Pull request closed without merging**